src/packages/utils/tests/json-rpc-data.test.ts

```diff
@@ -0,0 +1,187 @@
+import assert from "assert";
+import {Data} from "..";
+
+describe("json-rpc-data", () => {
+  const validValues = [ "0x", "0x1", "0x1234", Buffer.from([]), Buffer.from([0x12,0x34]) ];
+  const invalidValues: any[] = [ "1234", 1234n, NaN/*, undefined, null, 1234, [], {}, "0x-1234"*/ ]; // todo: this should be addressed in rewrite of json-rpc-data
+                                                                                                     // See related https://github.com/trufflesuite/ganache/issues/2728
+  const validBytelengths = (() => { let i = 0; return [...new Array(100)].map(_ => i++); })(); // [0...99]
+  const invalidBytelengths: any[] = [ -1, "1", {}, [], null, NaN ];
+  const inputOf32Bytes = "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef";
+
+  function getExpectedString(value: Buffer|string, bytelength?: number) {
+    let expected: string;
+
+    if (typeof value === "string") {
+      expected = value.slice(2);
+    } else if (Buffer.isBuffer(value)) {
+      expected = (<Buffer>value).toString("hex");
+    } else {
+      throw new Error(`Type not supported ${typeof value}`)
+    }
+    if (bytelength !== undefined) {
+      const padCharCount = (bytelength - expected.length / 2) * 2; // (desired byte count - actual byte count) * 2 characters per byte
+      if (padCharCount > 0) {
+        expected = "0".repeat(padCharCount) + expected;
+      } else {
+        expected = expected.slice(Math.abs(padCharCount));
+      }
+    }
+    return "0x" + expected;
+  }
+
+  describe("ctor", () => {
```

Comment:
I think I'm a bit slow on this one, what's ctor?

Reply:
Ahhh, constructor! Let's spell it out for people like me, haha
```suggestion
  describe("constructor", () => {
```